src/Sulu/Bundle/MediaBundle/Media/Storage/S3Storage.php

```diff
@@ -32,9 +32,11 @@ class S3Storage extends FlysystemStorage
      */
     private $bucketName;
 
-    public function __construct(FilesystemInterface $filesystem, int $segments)
+    private $url;
+
+    public function __construct(FilesystemInterface $filesystem, int $segments, $url)
```

Comment:
For BC reasons new params require to be optional:

```suggestion
    /**
     * @var string|null
     */
    private $url;

    public function __construct(FilesystemInterface $filesystem, int $segments, ?string $url = null)
```